iree/test/model_benchmarks/CMakeLists.txt

```diff
@@ -9,68 +9,46 @@
 #                                                                              #
 # Benchmark models                                                             #
 #                                                                              #
-# All the lists should have the same number of elements. Each list describes   #
-# one aspect of the model. Elements at the same index are for the same model.  #
-#                                                                              #
-# Normally models to be benchmarked should be placed here becuase all          #
-# benchmark cases will be enabled for them. There might exist cases where we   #
-# cannot enable all the benchmark cases for one model; then they should be     #
-# placed directly in the cmake function calls in the next section.             #
+# Each module specification should be a list that contains the following       #
+# fields: MODULE_NAME, MODULE_TAGS, MODULE_SOURCE, MLIR_SOURCE,                #
+# ENTRY_FUNCTION, FUNCTION_INPUTS. See iree_mlir_benchmark_suite definition    #
+# for details about these fields.                                              #
 #                                                                              #
 ################################################################################
 
-set(BENCHMARK_MODULE_NAMES
-  "MobileNetV2"
-  "MobileNetV3Small"
-)
-
-# Each element is a comma-separated list.
-set(BENCHMARK_MODULE_TAGS
-  "fp32,imagenet"
-  "fp32,imagenet"
-)
-
-set(BENCHMARK_MODULE_SOURCES
-  "TensorFlow"
-  "TensorFlow"
+set(MOBILENET_V2_MODULE
+  "MobileNetV2"     # MODULE_NAME
+  "fp32,imagenet"   # MODULE_TAGS
+  "TensorFlow"      # MODULE_SOURCE
```

Comment:
can you remove MODULE_SOURCE? it looks like it's only used for creating a directory that shouldn't be needed - the name of the model should be unique enough (if we had a MobileNetV3Small that came from tf and one that came from tflite, we should name them MobileNetV3SmallTFLite etc)

Reply:
It's there for easy migration (as no directory structure and scripts will need to be changed) towards a future where we generate such model benchmarks directly from the Python source. `iree_mlir_benchmark_suite` should (naturally) generate benchmark suites in the "MLIR" category; in my original thoughts it's more meant for op microbenchmarks or small dispatch region benchmarks. Here we are reusing it also for model benchmarks before we have the infrastructure of Python benchmark artifact generation. After Stella revamps the framework side, `iree_mlir_benchmark_suite` will have `MODULE_SOURCE` removed and always generate into "MLIR" category. These models, if coming from TensorFlow, will implicitly have "TensorFlow" as the module source.

Also see my previous reply to your original comment here: https://github.com/google/iree/pull/6108#discussion_r646003643.

Hope this clarifies why it's there. :)

Reply:
I remain unconvinced that having a magic directory structure independent from the cmake definition and the model name is better than just having a unique name :)

We want to make it easy to reference tests - whether it's finding where they are defined, tracking them through tools/scripts, greping for them in megabytes of logs, inspecting them on dashboards, etc. Having the extra implicit layer in there just makes all that harder. There's also the issue of a model that comes from a different source being a different model - something that has run through the tflite compiler + tosa + tosa->linalg is going to be materially different than one that comes through saved model + hlo + hlo->linalg - especially when factoring in weird tflite things like post-training quantization. Having been on the dev side of big codebases like lingvo they also have totally different model definition paths based on what you are exporting.

I just am not seeing much but pain from trying to alias the same names but shoving them in different directories. If the directory was relative to where the benchmark was defined that would be different: for example, iree/test/model_benchmarks/tensorflow/ and iree/test/model_benchmarks/tflite/ that each have a CMakeLists.txt that generate to $cmakedir/iree/test/model_benchmarks/tensorflow/ and $cmakedir/iree/test/model_benchmarks/tflite/ is great - but having this single file fork off to different directories is spooky.

Reply:
Yeah, the dir/file name is really a reflection of the unwieldiness of the reality: many frameworks, many conversion tools, many compiler flags, many runtime flags, a combination of various paths. :) Charting a clean way is not that straightforward. I think we can sync more offline later. I was trying to say we might not even need the `MODULE_SOURCE` thing later so it's there for now to bridge. But I guess I cannot foretell how soon we can reach the nicer future. So I get your point here and here you go: 21d5057. :)